File: bindings/python/test/test_interval.py
import unittest
import sys
sys.path.insert(1, '.')
sys.path.insert(1, '..')
import cconfigspace as ccs
from math import sin


class TestInterval(unittest.TestCase):

  def test_create(self):
    i = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 1.0)
    self.assertEqual( ccs.NumericType.FLOAT, i.type )
    self.assertEqual( -1.0, i.lower )
    self.assertEqual( 1.0, i.upper )
    self.assertTrue( i.lower_included )
    self.assertFalse( i.upper_included )

  def test_empty(self):
    i = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 1.0)
    self.assertFalse( i.empty )
    i = ccs.Interval(t = ccs.NumericType.FLOAT, lower = 1.0, upper = -1.0)
    self.assertTrue( i.empty )

  def test_equal(self):
    i1 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 1.0)
    i2 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 1.0)
    self.assertEqual(i1, i2)
    i2 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 1.0, upper_included = True)
    self.assertNotEqual(i1, i2)

  def test_intersect(self):
    i1 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 1.0)
    i2 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = 0.0, upper = 2.0)
    i3 = i1.intersect(i2)
    i4 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = 0.0, upper = 1.0)
    self.assertEqual( i4, i3 )
    i1 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 0.0)
    i2 = ccs.Interval(t = ccs.NumericType.FLOAT, lower = 0.0, upper = 2.0)
    i3 = i1.intersect(i2)
    self.assertTrue( i3.empty )

  def test_contains(self):
    i = ccs.Interval(t = ccs.NumericType.FLOAT, lower = -1.0, upper = 1.0)
    self.assertTrue( i.contains(0.0) )
    self.assertFalse( i.contains(2.0) )
    i = ccs.Interval(t = ccs.NumericType.INT, lower = -5, upper = 5)
    self.assertTrue( i.contains(0) )
    self.assertFalse( i.contains(6) )

if __name__ == '__main__':
    unittest.main()

File: bindings/ruby/test/test_interval.rb
require 'minitest/autorun'
require_relative '../lib/cconfigspace'

class CConfigSpaceTestInterval < Minitest::Test
  def test_new
    i = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 1.0)
    assert_equal( :CCS_NUMERIC_TYPE_FLOAT, i.type)
    assert_equal(-1.0, i.lower)
    assert_equal( 1.0, i.upper)
    assert( i.lower_included? )
    refute( i.upper_included? )
  end

  def test_empty
    i = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 1.0)
    refute( i.empty? )
    i = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: 1.0, upper: -1.0)
    assert( i.empty? )
  end

  def test_equal
    i1 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 1.0)
    i2 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 1.0)
    assert_equal(i1, i2)
    i2 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 1.0, upper_included: true)
    refute_equal(i1, i2)
  end

  def test_intersect
    i1 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 1.0)
    i2 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: 0.0, upper: 2.0)
    i3 = i1.intersect(i2)
    i4 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: 0.0, upper: 1.0)
    assert_equal(i4, i3)
    i1 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 0.0)
    i2 = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: 0.0, upper: 2.0)
    i3 = i1.intersect(i2)
    assert(i3.empty?)
  end

  def test_include
    i = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_FLOAT, lower: -1.0, upper: 1.0)
    assert( i.include?(0.0) )
    refute( i.include?(2.0) )
    i = CCS::Interval::new(type: :CCS_NUMERIC_TYPE_INT, lower: -5, upper: 5)
    assert( i.include?(0) )
    refute( i.include?(6) )
  end

end
